Makefile.uk

```diff
@@ -0,0 +1,319 @@
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+################################################################################
+# Library registration
+################################################################################
+$(eval $(call addlib_s,liblibctest,$(CONFIG_LIBLIBCTEST)))
+
+################################################################################
+# Sources
+################################################################################]
+LIBLIBCTEST_PATCHDIR = $(LIBLIBCTEST_BASE)/patches
+LIBLIBCTEST_URL = git://nsz.repo.hu:49100/repo/libc-test
+LIBLIBCTEST_BRANCH = master
+
+$(eval $(call clone,liblibctest,$(LIBLIBCTEST_URL),$(LIBLIBCTEST_BRANCH),$(LIBLIBCTEST_PATCHDIR)))
+
+################################################################################
+# Helpers
+################################################################################
+LIBLIBCTEST_EXTRACTED = $(LIBLIBCTEST_ORIGIN)/src
+
+################################################################################
+# Library includes
+################################################################################
+LIBLIBCTEST_CINCLUDES-y   += -I$(LIBLIBCTEST_EXTRACTED)/common
+LIBLIBCTEST_CXXINCLUDES-y += -I$(LIBLIBCTEST_EXTRACTED)/common
+
+################################################################################
+# Flags
+################################################################################
+
+# Suppress some warnings to make the build process look neater
+LIBLIBCTEST_FLAGS_SUPPRESS =
+
+################################################################################
+# Library sources
+################################################################################
+LIBLIBCTEST_SRCS-y += $(LIBLIBCTEST_BASE)/include/main.c
+
+ifeq ($(filter y,$(CONFIG_LIBNOLIBC)),)
+    ifneq ($(filter y,$(CONFIG_LIBLIBCTEST_ACCESS_TESTS) $(CONFIG_LIBUKTEST_ALL)),)
+        LIBLIBCTEST_SRCS-y += $(LIBLIBCTEST_BASE)/access_test.c
+    endif
+endif
+
+ifneq ($(filter y,$(CONFIG_LIBLIBCTEST_STRING_TESTS) $(CONFIG_LIBUKTEST_ALL)),)
+	LIBLIBCTEST_SRCS-y += $(LIBLIBCTEST_EXTRACTED)/functional/string_memcpy.c
+    LIBLIBCTEST_SRCS-y += $(LIBLIBCTEST_EXTRACTED)/functional/string_strchr.c
+    LIBLIBCTEST_SRCS-y += $(LIBLIBCTEST_EXTRACTED)/functional/string_strcspn.c
+    LIBLIBCTEST_SRCS-y += $(LIBLIBCTEST_EXTRACTED)/functional/string_strstr.c
```

Comment:
As far as I could tell, there is no strstr implementation in nolibc so maybe this should be moved in the ifeq as well.

Reply:
Moved it.